nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBLookupService.java

```diff
@@ -223,7 +223,9 @@ protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
         return Collections.unmodifiableList(_temp);
     }
 
-    private Document findOne(Document query, Document projection) {
+    private Document findOne(Document query, Document projection,Map<String, String> context) {
+        final String databaseName = getProperty(DATABASE_NAME).evaluateAttributeExpressions(context).getValue();
+        final String collection = getProperty(COLLECTION_NAME).evaluateAttributeExpressions(context).getValue();
```

Comment:
The `databaseName` and `collection` properties are already set in `onEnabled()`.  Moving the property evaluation to this method should also remove setting the values in the `onEnabled()` method.

Reply:
Creating the new variables as you have makes sense, but the member variables for `databaseName` and `collection` should be removed, since these additions shadow the member variable names.

Reply:
okay thanks. I will  remove those member variables  and related statements from  onEnabled  method

Reply:
Sounds good!

Reply:
I have made the changes please have a look  thanks